### PR TITLE
Add gen_snapshot only for non debug versions of artifacts.

### PR DESCRIPTION
**Pull request by godofredoc**
artifacts.zip is expected to include multiple files in the debug configuration but only gen_snapshot in all the others. This PR is adding that logic to the archive gn target.

Bug: https://github.com/flutter/flutter/issues/81855
Bug: https://github.com/flutter/flutter/issues/112187

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: build/archives/BUILD.gn [hunk 1]
```diff
@@ -20,14 +20,18 @@ generated_file("artifacts_entitlement_config") {
...
     deps = [
       "//flutter/flutter_frontend_server:frontend_server",
       "//flutter/impeller/compiler:impellerc",
+      "//flutter/impeller/compiler:impellerc_license",
       "//flutter/impeller/tessellator:tessellator_shared",
       "//flutter/lib/snapshot:generate_snapshot_bin",
       "//flutter/shell/testing:testing",
+      "//flutter/tools/path_ops:path_ops",
+      "//flutter/tools/path_ops:path_ops_license",
+      "//third_party/dart/runtime/bin:gen_snapshot",
```
Comment (zanderso) [inline build/archives/BUILD.gn:34]: I think the existing build gets `gen_snapshot` for this from the target `//flutter/lib/snapshot` here https://github.com/flutter/engine/blob/main/lib/snapshot/BUILD.gn#L248. There seems to be some logic in that file for building `gen_snapshot` that goes beyond a simple dependency on `//third_party/dart/runtime/bin:gen_snapshot`, so I'm a little worried that some differences might creep in here.

Comment (godofredoc) [reply build/archives/BUILD.gn:34]: After changing to `//flutter/lib/snapshot` gn complains about `//flutter/lib/snapshot` not generating the gen_snapshot binary.

```
The file:
  //out/host_profile/gen_snapshot
is listed as an input or source for the target:
  //flutter/build/archives:artifacts
but this file was not generated by any dependencies of the target. The target
that generates the file is:
  //third_party/dart/runtime/bin:gen_snapshot
```

Comment (zanderso) [reply build/archives/BUILD.gn:34]: I think the issue there is that the tool for the `compiled_action` isn't listed as a public dependency, but I took a closer look, and I don't think the logic from `//flutter/lib/snapshot` is needed here. That target needs special logic for producing a `gen_snapshot` that can run on a 64-bit intel host and produce a 32-bit intel snapshot, which we don't actually ship. So this PR lgtm.

**Review by zanderso — APPROVED** [review]
lgtm

**Comment by godofredoc** [comment]
> lgtm

Thanks @zanderso

**Pull request merged**